src/lib.rs

```diff
@@ -137,19 +151,389 @@ pub fn get_verifiable_credential_public_key(
+    #[serde(rename = "arsInfos")]
+    pub ars_infos: HashMap<ArIdentity, AnonymityRevokerInfo>,
+    #[serde(rename = "idObject")]
+    pub id_object: IdentityObject,
+    #[serde(rename = "revealedAttributes")]
+    pub revealed_attributes: Vec<u8>,
+    #[serde(rename = "credNumber")]
+    pub cred_number: u8,
+    #[serde(rename = "idCredSec")]
+    pub id_cred_sec_hex: String,
+    #[serde(rename = "prfKey")]
+    pub prf_key_hex: String,
+    #[serde(rename = "blindingRandomness")]
+    pub blinding_randomness_hex: String,
+    #[serde(rename = "attributeRandomness")]
+    pub attribute_randomness_hex: HashMap<String, String>,
+    #[serde(rename = "credentialPublicKeys")]
+    pub credential_public_keys_hex: CredentialPublicKeysHex,
+}
+
+#[derive(Debug, Serialize)]
+pub struct IdentityObject {
+    #[serde(rename = "preIdentityObject")]
+    pub pre_identity_object: PreIdentityObject,
+    #[serde(rename = "attributeList")]
+    pub attribute_list: AttributeList,
+    #[serde(rename = "signature")]
+    pub signature_hex: String,
+}
+
+#[derive(Debug, Serialize)]
+pub struct PreIdentityObject {
+    #[serde(rename = "idCredPub")]
+    pub id_cred_pub_hex: String,
+    #[serde(rename = "ipArData")]
+    pub ip_ar_data: HashMap<ArIdentity, ArData>,
+    #[serde(rename = "choiceArData")]
+    pub choice_ar_data: ChoiceArParameters,
+    #[serde(rename = "idCredSecCommitment")]
+    pub id_cred_sec_commitment_hex: String,
+    #[serde(rename = "prfKeyCommitmentWithIP")]
+    pub prf_key_commitment_with_ip_hex: String,
+    #[serde(rename = "prfKeySharingCoeffCommitments")]
+    pub prf_key_sharing_coeff_commitments_hex: Vec<String>,
+    #[serde(rename = "proofsOfKnowledge")]
+    pub proofs_of_knowledge_hex: String,
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+pub struct ChoiceArParameters {
+    #[serde(rename = "arIdentities")]
+    pub ar_identities: Vec<ArIdentity>,
+    #[serde(rename = "threshold")]
+    pub threshold: u32,
+}
+
+#[derive(Debug, Serialize)]
+pub struct ArData {
+    #[serde(rename = "encPrfKeyShare")]
+    pub enc_prf_key_share_hex: String,
+    #[serde(rename = "proofComEncEq")]
+    pub proof_com_enc_eq_hex: String,
+}
+
+#[derive(Debug, Serialize)]
+pub struct AttributeList {
+    #[serde(rename = "validTo")]
+    pub valid_to_year_month: String,
+    #[serde(rename = "createdAt")]
+    pub created_at_year_month: String,
+    #[serde(rename = "maxAccounts")]
+    pub max_accounts: u8,
+    #[serde(rename = "chosenAttributes")]
+    pub chosen_attributes: HashMap<String, String>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct CredentialPublicKeys<Key> {
+    #[serde(rename = "keys")]
+    pub keys: HashMap<KeyIndex, Key>,
+    #[serde(rename = "threshold")]
+    pub threshold: u8,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct VerifyKeyWithScheme {
+    #[serde(rename = "schemeId")]
+    pub scheme_id: String,
+    #[serde(rename = "verifyKey")]
+    pub key_hex: String,
+}
```

Comment:
It only exists because the JSON serialization of `AccountCredential` uses it (via `CredentialPublicKeysWithScheme`). Unlike in the input of the request function where it's just a string. But from what you say I understand that you could also provide the tagged variant as input?

Reply:
That worked, consolidated the types.